syntax = "proto3";

package transport_catalogue_serialize;

message Rgb {
  uint32 red = 1;
  uint32 green = 2;
  uint32 blue = 3;
}

message Rgba {
  uint32 red = 1;
  uint32 green = 2;
  uint32 blue = 3;
  double opacity = 4;
}

message Color {
  oneof color
  {
   string string_value  = 1;
   Rgb rgb_value = 2;
   Rgba rgba_value = 3;
   }
}